src/com/amazon/corretto/crypto/provider/AesCbcSpi.java

```diff
@@ -137,19 +138,34 @@ private enum CipherState {
   }
 
   private boolean noPadding() {
-    return padding == NO_PADDING;
+    return paddingScheme.equals(Padding.NONE);
   }
 
   @Override
   protected void engineSetMode(final String mode) throws NoSuchAlgorithmException {
-    // no op. One only needs to provide an implementation if the same Spi class instance can be used
-    // for different modes.
+    if (!"CBC".equalsIgnoreCase(mode)) {
+      throw new NoSuchAlgorithmException("Only CBC mode is supported.");
+    }
   }
 
   @Override
   protected void engineSetPadding(final String padding) throws NoSuchPaddingException {
-    // no op. One only needs to provide an implementation if the same Spi class instance is used for
-    // different paddings.
+    if (padding == null) {
+      throw new NoSuchPaddingException("Padding cannot be null.");
+    }
+    if (!inputIsEmpty) {
+      throw new NoSuchPaddingException("Padding cannot be set during an operation.");
+    }
+    Predicate<String> paddingPredicate = n -> n.split("/")[2].equalsIgnoreCase(padding);
```

Comment:
dereferencing the `split(...)` result without checking result seems like a recipe for NPE for malformed `padding` String inputs. can we add a check?

Reply:
I made the sets immutable. I also tried to refactor it to avoid expanding out all the name combinations, but it looks more annoying than expected (have to dig into all the ways the JDK passes in these names).